locator/abstract_replication_strategy.hh

```diff
@@ -128,11 +128,48 @@ public:
     future<dht::token_range_vector> get_pending_address_ranges(const token_metadata_ptr tmptr, std::unordered_set<token> pending_tokens, inet_address pending_address, locator::endpoint_dc_rack dr) const;
 };
 
+using replication_strategy_ptr = seastar::shared_ptr<const abstract_replication_strategy>;
+using mutable_replication_strategy_ptr = seastar::shared_ptr<abstract_replication_strategy>;
```

Comment:
let's split the above into a separate patch.
it seems orthogonal to reintroducing an abstract effective_replication_map

Reply:
Ok